OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -1381,15 +1402,13 @@ static class DMOpcodeHandlers {
 
             DreamList containerList;
             if (container != null && container.IsSubtypeOf(DreamPath.Atom)) {
-                containerList = container.GetVariable("contents").GetValueAsDreamList();
+                container.GetVariable("contents").TryGetValueAsDreamList(out containerList);
             } else {
                 containerList = container as DreamList;
             }
 
-            if (value.TryGetValueAsString(out string refString)) {
-                int refID = int.Parse(refString);
-
-                state.Push(new DreamValue(DreamObject.GetFromReferenceID(state.DreamManager, refID)));
+            if (value.TryGetValueAsString(out string refString) && int.TryParse(refString, out var refId)) {
```

Comment:
BYOND seems to return null if the string is not a valid ref ID, even if the string appears in the list.

Reply:
"valid ref ID" includes tags here, not just ref numbers.